testcases/OpTestEEH.py

```diff
@@ -379,3 +204,170 @@ def check_eeh_pe_recovery(self, pe):
+        pci_domains = self.cv_HOST.host_get_list_of_pci_domains()
+        print "Skipping the root phb %s for fenced PHB Testcase" % root_domain
+        pci_domains.remove(root_domain)
+        self.cv_HOST.host_gather_opal_msg_log()
+        self.cv_HOST.host_gather_kernel_log()
+        l_con = self.cv_SYSTEM.sys_get_ipmi_console()
+        self.cv_IPMI.ipmi_host_login(l_con)
+        self.cv_IPMI.ipmi_host_set_unique_prompt()
+        l_con.run_command("stty cols 300")
+        l_con.run_command("stty rows 10")
+        l_con.run_command("dmesg -D")
+        l_con.run_command("uname -a")
+        l_con.run_command("cat /etc/os-release")
+        for domain in pci_domains:
+            self.prepare_logs()
+            cmd = "echo 0x8000000000000000 > /sys/kernel/debug/powerpc/%s/err_injct_outbound; lspci;" % domain
+            print "=================Injecting the fenced PHB error on PHB: %s=================" % domain
+            l_con.run_command(cmd)
+            # Give some time to EEH PCI Error recovery
+            time.sleep(30)
+            l_con.sendline("\r\n")
+            self.gather_logs()
+            if not self.check_eeh_phb_recovery(domain):
+                msg = "PHB domain %s recovery failed" % domain
+                raise OpTestError(msg)
+            else:
+                print "PHB %s recovery successful" % domain
+
+class OpTestEEHmax_fenced_phb(OpTestEEH):
+    ##
+    # @brief  This testcase has below steps
+    #         1. Get the list of pci PHB domains
+    #         2. Get the root PHB domain where the root file system
+    #            is installed(We need to skip this as EEH recovery will
+    #            fail on root PHB).
+    #         3. Set the MAX EEH Freeze count to 1 so that we can
+    #            test max EEH Recovery capacity within less for loop executions
+    #            By default it is 6,
+    #         4. Start injecting the fenced PHB errors two times across all PHB domains
+    #            except root PHB.( As we set max EEH Freeze count to 1).
+    #            So expectation is first time it should recover and second time
+    #            EEH should properly remove the device and OS should not crash.
+    #
+    # @return BMC_CONST.FW_SUCCESS or BMC_CONST.FW_FAILED
+    #
+    def runTest(self):
+        root_domain = self.cv_HOST.host_get_root_phb()
+        pci_domains = self.cv_HOST.host_get_list_of_pci_domains()
+        print "Skipping the root phb %s for fenced PHB Testcase" % root_domain
+        pci_domains.remove(root_domain)
+        self.cv_HOST.host_gather_opal_msg_log()
+        self.cv_HOST.host_gather_kernel_log()
+        # Set the max EEH freeze count to 1
+        cmd = "echo 1 > /sys/kernel/debug/powerpc/eeh_max_freezes"
+        self.cv_HOST.host_run_command(cmd)
+        l_con = self.cv_SYSTEM.sys_get_ipmi_console()
+        self.cv_IPMI.ipmi_host_login(l_con)
+        self.cv_IPMI.ipmi_host_set_unique_prompt()
```

Comment:
I think I fixed it.... although I could be hitting the issue where I should be rebooting teh host inbetween each of the EEH tests (as we were just discussing on Slack)